### ISSUE TITLE
Simple Windows compilation issues

### ISSUE THREAD

**Issue by osrf-migration** [ignitionrobotics/ign-msgs#26]
**Original report ([archived issue](https://osrf-migration.github.io/ignition-gh-pages/#!/ignitionrobotics/ign-msgs/issues/26)) by Renato Cherullo (Bitbucket: [rcherullo](https://bitbucket.org/%7Bec5c23d3-54b0-4b3e-a5ed-8ea19367e5f3%7D/)).**

----------------------------------------
Hi all,

While compiling for Windows in VS2017, I get the following errors:

```
#!

4>C:\gz-ws\osrf\ignition-msgs1\tools\ign_TEST.cc(29): error C3861: 'popen': identifier not found [C:\gz-ws\build\osrf\ignition-msgs1\tools\UNIT_ign_TEST.vcxproj]

4>C:\gz-ws\osrf\ignition-msgs1\tools\ign_TEST.cc(43): error C3861: 'pclose': identifier not found [C:\gz-ws\build\osrf\ignition-msgs1\tools\UNIT_ign_TEST.vcxproj]

4>C:\gz-ws\osrf\ignition-msgs1\tools\ign_TEST.cc(93): error C3861: 'setenv': identifier not found [C:\gz-ws\build\osrf\ignition-msgs1\tools\UNIT_ign_TEST.vcxproj]
```

The fix is quite simple, maybe I could create a pull request for it: popen and pclose can be replaced with _popen and _pclose, while setenv can be replaced with _putenv_s. This works and the tests pass.

The question is how to implement this platform check? #ifdefs? Extract those methods to another file and do the checks there? Looking forward for your guidance.

  []s Renato C.

**Comment by chapulina** [comment]
Has this been fixed? Or is this an issue with some specific Windows / Visual Studio version? Our Windows CI has been passing:

https://build.osrfoundation.org/job/ignition_msgs-ci-ign-msgs4-windows7-amd64/

**Comment by scpeters** [comment]
this was fixed in 5d8880376d0d21f82fac0e3caa28ee4527476503
